src/dialogs/examples/basic/PopupSignInDialog.ts

```diff
@@ -0,0 +1,51 @@
+import * as builder from "botbuilder";
+import { TriggerActionDialog } from "../../../utils/TriggerActionDialog";
+import { DialogIds } from "../../../utils/DialogIds";
+import { DialogMatches } from "../../../utils/DialogMatches";
+import * as config from "config";
+
+export class PopupSignInDialog extends TriggerActionDialog {
+
+    private static async PopUpSignIn(session: builder.Session, args?: any | builder.IDialogResult<any>, next?: (args?: builder.IDialogResult<any>) => void): Promise<void> {
+        let cards = new Array<builder.HeroCard>();
+
+        let buttons = new Array<builder.CardAction>();
+            /**
+             *  This is PopUp SignIn Dialog Class.
+             *  main purpose of this class is to Display the PopUp SignIn Card
+             */
+
+            // let input = "";
+            // if (args.response) {
+            //     input = args.response;
+            // }
+
+        let popUpUrl = config.get("app.baseUri") + "/botViews/popUpSignin.html";
+
+        buttons.push(new builder.CardAction(session)
+            .type("signin")
+            .title("Sign In")
```

Comment:
@jotrick is there a principle we follow in this project for when a string goes in index.json vs. inline in the ts file?

Reply:
@jotrick has used index.json for string constant, I have used the same now.